### DISCUSSION TITLE
(Reinstalling Obsidian with the newest Installer solved the problem) Execute command has a few seconds delay, try reuse a shell ChildProcess to execute command.

### DISCUSSION THREAD

**Discussion by LudwigWS** [Taitava/obsidian-shellcommands#227] [Ideas]
Running on my macOS (12.4, MacBook Pro 2019 16'), whatever I run a command, even a `echo 'hello'`, it has a few seconds delay about 2-3s that is annoying.

Are you the same, how long time should it take for you?

I think the time is spent on creating a new process that is slow,  I try to create a shell ChildProcess at the Plugin loading or the first command running, use it to execute command rather than create a new ChildProcess to execute a single command every time.

```javascript
this.shell = spawn("bash"); // only create one time.

let cmd = "/usr/local/bin/emacsclient -n /tmp/text.txt\n" // don't forget \n
this.shell.stdin.write(cmd); // send the command string to stored shell process
```

this may speed up the execution.

**Comment by LudwigWS** [comment]
It seems that the wording of my post is a bit impolite, I’m sorry about this. Because my English is poor, It's difficult for me to express what I mean correctly. I have wrote a simple plugin to implement feature of myself. I'm not asking you to do anything, just a suggestion. Thank your for your working.

**Reply by Taitava** [reply]
Oh, I didn't take it impolitely, don't worry! 🙂 I haven't responded yet as I feel I might be coming sick (flu), so I decided to answer later.

As you are using macOS, there should be no delay. I don't have macOS myself, but I've had discussions with Mac users and most of them haven't reported about a delay. If you would use Windows, then there is a delay with PowerShell, but that's not the case here.

There has been one Mac user with a delay problem: https://github.com/Taitava/obsidian-shellcommands/discussions/187 . I don't know the exact details about what actually caused this, but @imeed166 mentioned a theme they used caused the delay. Maybe you could try to temporarily disable all other community plugins and switch to the default theme, just to see if any of those can cause the problem. This is just guessing.

Changing the execution mechanism can be considered, but it will take some time for me to figure out all the consequences of it.

**Reply by LudwigWS** [reply]
I switch to a new vault and use the default theme, but delay still exist. It's strange that you don't have delay, I had supposed that it's a problem of Electron framework.

I'm sorry to hear that you are feeling uncomfortable, put away computer now, take more rest please. I hope you get better soon.

**Reply by Taitava** [reply]
> I'm sorry to hear that you are feeling uncomfortable, put away computer now, take more rest please. I hope you get better soon.

Thank you, I rested a lot for a couple of days, which was enough, the flu didn't start after all, which was good 🙂 . I've learned that I need to rest a lot if I feel any symptoms that a flu might be coming, so I can sometimes avoid it. 👍

**Comment by LudwigWS** [comment]
I had tested that even I started a new plugin, even only with one line code `child_process.exec("echo 'hello'")`, it still have delay, so the problem seem to be Obsidian's, we had better wait for more Mac users's response to determine if it's a common problem.

**Reply by imed-ghomari** [reply]
I'm on macOS 12.3.1 with obsidian 0.14.6 installer 0.14.5, no delay. What version of obsidian are you using ?

**Reply by LudwigWS** [reply]
macOS 12.4, Obsidian is 0.14.6 installer 0.14.2, it doesn't make sense...

**Reply by LudwigWS** [reply]
> I'm on macOS 12.3.1 with obsidian 0.14.6 installer 0.14.5, no delay. What version of obsidian are you using ?

Reinstall Obsidian have solved my problem, the same version 0.14.6, but different installer. I just wanted to give it a try, but it did work to my surprise.

**Reply by imed-ghomari** [reply]
I was going to suggest that, glad it worked out.

**Reply by Taitava** [reply]
Nice to hear that it's solved now! 🙂
